File: sdk/samples/all in on code/Visual Studio 2010/VBAzureBingMaps/UnitTest/DataServiceTest.vb
Imports System.Text
Imports System.Collections.Generic
Imports System.Linq
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports AzureBingMaps.UnitTest.TravelDataServiceReference

<TestClass()> _
Public Class DataServiceTest
    <TestMethod()> _
    Public Sub AddToTravelTest()
        Dim ctx As New TravelDataServiceContext(New Uri("http://localhost:28452/DataService/TravelDataService.svc"))
        Dim travel As New Travel() With { _
          .PartitionKey = "Lanxing", _
          .RowKey = Guid.NewGuid(), _
          .Place = "Shanghai", _
          .Latitude = 31, _
          .Longitude = 121, _
          .Time = DateTime.Now _
        }
        ctx.AddObject("Travels", travel)
        ctx.SaveChanges()
    End Sub

    <TestMethod()> _
    Public Sub UpdateTravelTest()
        Dim ctx As New TravelDataServiceContext(New Uri("http://localhost:28452/DataService/TravelDataService.svc"))
        Dim travel As Travel = ctx.Travels.First()
        travel.Place = "New York"
        ctx.UpdateObject(travel)
        ctx.SaveChanges()
    End Sub

    <TestMethod()> _
    Public Sub DeleteTravelTest()
        Dim ctx As New TravelDataServiceContext(New Uri("http://localhost:28452/DataService/TravelDataService.svc"))
        Dim travel As Travel = ctx.Travels.First()
        ctx.DeleteObject(travel)
        ctx.SaveChanges()
    End Sub
End Class

File: sdk/samples/all in on code/Visual Studio 2010/CSAzureBingMaps/UnitTest/DataServiceTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using AzureBingMaps.UnitTest.TravelDataServiceReference;

namespace AzureBingMaps.UnitTest
{
    [TestClass]
    public class DataServiceTest
    {
        [TestMethod]
        public void AddToTravelTest()
        {
            TravelDataServiceContext ctx = new TravelDataServiceContext
                (new Uri("http://localhost:28452/DataService/TravelDataService.svc"));
            Travel travel = new Travel()
            {
                PartitionKey = "Lanxing",
                RowKey = Guid.NewGuid(),
                Place = "Shanghai",
                Latitude = 31,
                Longitude = 121,
                Time = DateTime.Now
            };
            ctx.AddObject("Travels", travel);
            ctx.SaveChanges();
        }

        [TestMethod]
        public void UpdateTravelTest()
        {
            TravelDataServiceContext ctx = new TravelDataServiceContext
                (new Uri("http://localhost:28452/DataService/TravelDataService.svc"));
            Travel travel = ctx.Travels.First();
            travel.Place = "New York";
            ctx.UpdateObject(travel);
            ctx.SaveChanges();
        }

        [TestMethod]
        public void DeleteTravelTest()
        {
            TravelDataServiceContext ctx = new TravelDataServiceContext
                (new Uri("http://localhost:28452/DataService/TravelDataService.svc"));
            Travel travel = ctx.Travels.First();
            ctx.DeleteObject(travel);
            ctx.SaveChanges();
        }
    }
}
